modules/heat_conduction/src/materials/SemiconductorLinearConductivity.C

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SemiconductorLinearConductivity.h"
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<SemiconductorLinearConductivity>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredCoupledVar("temp", "variable for temperature");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.addRequiredParam<Real>("sh_coeff_A", "Steinhart_Hart coefficient A of the material");
+  params.addRequiredParam<Real>("sh_coeff_B", "Steinhart_Hart coefficient B of the material");
+  return params;
+}
+
+SemiconductorLinearConductivity::SemiconductorLinearConductivity(const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _sh_coeff_A(getParam<Real>("sh_coeff_A")),
+    _sh_coeff_B(getParam<Real>("sh_coeff_B")),
+    _T(coupledValue("temp")),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _electric_conductivity(declareProperty<Real>(_base_name + "electrical_conductivity")),
+    _delectric_conductivity_dT(declarePropertyDerivative<Real>(
+        _base_name + "electrical_conductivity", getVar("temp", 0)->name()))
+{
+}
+
+void
+SemiconductorLinearConductivity::computeQpProperties()
+{
+  _electric_conductivity[_qp] = exp((_sh_coeff_A - 1 / _T[_qp]) / _sh_coeff_B);
```

Comment:
I think that test is failing on a divide by zero here because the temperature is initially zero. Add code like this at the beginning of this function to ensure that temperatures are always positive:
```
if (MooseUtils::absoluteFuzzyLessEqual(_T[_qp], 0.0))
  mooseError("Encountered zero or negative temperature in SemiconductorLinearConductivity");
```
In your test input file, add something like:
```
  initial_condition = 400
```
in the `T` variable block to avoid hitting that condition.

Reply:
Also, we encourage the use of mooseAssert(). Assert the conditions that must be true for your function not to crash. If you can't divide by zero, assert that!

Reply:
@permcody Since mooseAssert() only has an effect on the debug executable, I don't like using it in places like this where you could easily trip the assert due to user input. I would like to see mooseError() get called if there is a negative temperature.

Reply:
Completely agree. I wasn't suggesting instead. I was encouraging adding the assert "in addition to". As you pointed out, it's zero cost in production runs and can help track down floating point problems in debug/devel modes.

Reply:
@yeh512 The change I suggested for making `sh_coeff_B` range-checked would prevent from dividing by zero for that, but you still need a divide by zero check here on `_T[_qp]`.  Just use the code I provided in my comment above for that.

@permcody I'm not sure what the point would be of having both an assert and an error for the same thing, since the error would always catch it. Are you suggesting using an assert for checking something else?